packages/typespec-client-generator-core/test/types.test.ts

```diff
@@ -2027,6 +2027,87 @@ describe("typespec-client-generator-core: types", () => {
         code: "@azure-tools/typespec-client-generator-core/conflicting-multipart-model-usage",
       });
     });
+    it("multipart combined with shared route should not report conflicting model usage", async function () {
+      const diagnostics = await runner.diagnose(
+        `
+        @service({title: "Test Service"}) namespace TestService;
+        model MultiPartRequest {
+          id: string;
+          profileImage: bytes;
+        }
+  
+        @sharedRoute
+        @post op multipartUse(@header contentType: "multipart/form-data", @body body: MultiPartRequest): NoContentResponse;
+        @sharedRoute
+        @post op multipartUseAsString(@body body: MultiPartRequest): string;
+      `
+      );
+      getAllModels(runner.context);
+      expectDiagnostics(diagnostics, {});
+    });
+    it("multipart combined with shared route and core template should not report conflicting model usage", async function () {
```

Comment:
But if it's different core template, it will report conflicting model usage, which it should not.